### PR TITLE
Add new teleconsult numbers to weekly report

### PR DESCRIPTION
**Pull request by vkrmis**
**Story card:** [ch1647](https://app.clubhouse.io/simpledotorg/story/1647/add-telemed-mvp-data-to-the-weekly-telemed-report)

## Because

We want to continue tracking Telemed progress via the existing weekly report till we completely transition to the MVP

## This addresses

Adds some fields to show Telemed MVP data in the weekly report

### CODE REVIEW

File: lib/tasks/scripts/telemedicine_reports.rb [hunk 1]
```diff
@@ -84,6 +84,9 @@ def assemble_report_data
       "Patients with High Blood Sugar",
       "Patients with High BP or Sugar",
       "Teleconsult Button Clicks",
+      "Teleconsultation MVP Requests",
```
Comment (harimohanraj89) [inline lib/tasks/scripts/telemedicine_reports.rb:87]: Do we want to call it "MVP" in the report? Is there some other name that this report's consumers would prefer? I can't think of any off the top of my head, but we could check with Mahima.

File: lib/tasks/scripts/telemedicine_reports.rb [hunk 2]
```diff
@@ -120,7 +126,10 @@ def assemble_report_data
           district[:telemed_data][:high_bs],
           district[:telemed_data][:high_bp_or_bs],
           telemed_clicks,
-          percentage(telemed_clicks, district[:telemed_data][:high_bp_or_bs])
+          district[:telemed_data][:teleconsultation_requests],
+          district[:telemed_data][:teleconsultation_records],
+          district[:telemed_data][:teleconsultation_marked_completed],
+          percentage(telemed_clicks + district[:telemed_data][:teleconsultation_requests], district[:telemed_data][:high_bp_or_bs])
```
Comment (harimohanraj89) [inline lib/tasks/scripts/telemedicine_reports.rb:132]: This is awesome. We can safely assume that total teleconsults = green button clicks plus DB records? Does the MVP's green button clicks _not_ feed into the `telemed_clicks` number?

File: lib/tasks/scripts/telemedicine_reports.rb [hunk 3]
```diff
@@ -293,11 +326,20 @@ def facility_measures(facility, p_start, p_end)
     visits = (bps + sugars + appointments + drugs).uniq { |record| record[:patient_id] }
     high_bps = high_bps(bps)
     high_sugars = high_sugars(sugars)
+    teleconsult_mvp_requests = facility
+      .teleconsultations
+      .where("device_created_at >= ? AND device_created_at <= ?", p_start, p_end)
+      .select("DISTINCT(patient_id)")
+    teleconsult_mvp_records = teleconsult_mvp_requests.where.not(recorded_at: nil)
+    teleconsult_mvp_marked_completed = teleconsult_mvp_requests.where(request_completed: "yes")
```
Comment (harimohanraj89) [inline lib/tasks/scripts/telemedicine_reports.rb:334]: This is gonna be 0 for now, right?

File: spec/lib/tasks/scripts/telemedicine_reports_spec.rb [hunk 4]
```diff
@@ -11,6 +11,7 @@
   let!(:user_2) { create(:user, id: "31f4b6c4-6172-499a-95e1-bbbbbbbbbbbb", registration_facility: facility_2) }
   let!(:high_bps) { create_list(:blood_pressure, 2, :hypertensive, :with_encounter, facility: facility_1, user: user_1, recorded_at: period_start + 1.day) }
   let!(:high_bps_2) { create_list(:blood_pressure, 2, :with_encounter, :hypertensive, facility: facility_2, user: user_2, recorded_at: period_start + 1.day) }
+  let!(:teleconsultation_request) { create(:teleconsultation, facility: facility_1, patient: high_bps.first.patient, requester: user_1, requested_medical_officer: user_2, medical_officer: user_2, device_created_at: Date.parse("2020-08-04").beginning_of_day) }
```
Comment (harimohanraj89) [inline spec/lib/tasks/scripts/telemedicine_reports_spec.rb:14]: Could we make another couple TC requests? One with `nil` record info, one with "requester" `completed: true`?

**Review by harimohanraj89 — CHANGES_REQUESTED** [review]
Some questions and suggestions below!

File: lib/tasks/scripts/telemedicine_reports.rb (hunk 2 above)
Comment (vkrmis) [reply lib/tasks/scripts/telemedicine_reports.rb:132]: I'm going to check with Vinay to confirm, don't know this for sure.

File: spec/lib/tasks/scripts/telemedicine_reports_spec.rb (hunk 4 above)
Comment (vkrmis) [reply spec/lib/tasks/scripts/telemedicine_reports_spec.rb:14]: I'm going off of [this request](https://simpledotorg.slack.com/archives/CV9V1DWLV/p1602604499062900?thread_ts=1602600744.062700&cid=CV9V1DWLV) from Dr Reena atm, we can add more columns iteratively if they become valuable.

Comment (harimohanraj89) [reply spec/lib/tasks/scripts/telemedicine_reports_spec.rb:14]: Oh, sorry, that's not what I meant. I meant, for this test case, can we make more TC request records with different combinations of data, so we can ensure we're putting the right number into the column?

Comment (vkrmis) [reply spec/lib/tasks/scripts/telemedicine_reports_spec.rb:14]: Ah, took me a while, even after reading this clarification 😅. I'll do this.

File: lib/tasks/scripts/telemedicine_reports.rb (hunk 2 above)
Comment (vkrmis) [reply lib/tasks/scripts/telemedicine_reports.rb:132]: The `telemed_clicks` includes MVP clicks as well. Will change this accordingly.

File: lib/tasks/scripts/telemedicine_reports.rb (hunk 1 above)
Comment (vkrmis) [reply lib/tasks/scripts/telemedicine_reports.rb:87]: Briefed Mahima on this, she'll take a look and get back to me in the next few days.

File: spec/lib/tasks/scripts/telemedicine_reports_spec.rb (hunk 4 above)
Comment (harimohanraj89) [reply spec/lib/tasks/scripts/telemedicine_reports_spec.rb:14]: Man, I'm terrible at clarification 🤦 Thanks!

**Review by harimohanraj89 — APPROVED** [review]
Looks slick, copy is super clear 👍

**Pull request merged**